### DISCUSSION TITLE
Use shortcodes to return a choice value in `arg` immidiately

### DISCUSSION THREAD

**Discussion by davelosert** [johnlindquist/kit#557] [Ideas]
Hi there 👋🏻 ,

first of all: I really love this tool as it makes automating workflows sooo easy, it's incredible. So: Great work! ❤️ 

One little idea I had that would be great in my workflows: Having the abilty to make `arg` return the value of a choice with a single hotkey or shortcode immidiately. Kind of like the emacs or neovims `which-key` plugin.

Forgive me if this was already proposed or already works somehow, I looked through all ideas and docs and couldn't quite find it (the closest to it maybe being https://github.com/johnlindquist/kit/discussions/133).

I found out that this kind of works already (even though it is not documented for `arg` yet) like the following:

```typescript
const selected = await arg('[y]es or [n]o?', [
  { name: 'Yes', value: 'yes', shortcode: ['y'] },
  { name: 'No', value: 'no', shortcode: ['n'] }
]);
```

With this, I can now quick select either option with `y<space>` or `n<space>`.

However, I'd love to not have to hit `<space>` but just return the value from `arg` immidiately after hitting either `y` or `n`.

Of course, this would mean the options are not filterable anymore, so this kind of `arg` would have to be configurable. Maybe something like:

```typescript
const selected = await arg({
  // I found thise 'mode' setting in the types already - not sure what it does though :D
  // But it could be used to enable this behaivour
  mode: 'HOTKEY',
  placeholder: '[y]es or [n]o or [m]aybe.'
},
[
  { name: '[y]es', value: 'yes', shortcode: ['y'] },
  { name: '[n]o', value: 'no', shortcode: ['n'] },
  { name: '[m]aybe', value: 'no', shortcode: ['m'] }
]);
```

What do you think?

If you are open for PRs, I'd be happy to send one.

**Comment by johnlindquist** [comment]
I love the way you think! 😁 

I'm not quite open to PRs yet, although I'd be happy to help you set up a dev environment this week. I'll put some instructions together tomorrow.

I like the convention you suggested:

name: [B]anana

Where "b" would automatically select the choice.

I'll figure out how to get that into the app. You can get the general idea of input handling here:

https://github.com/johnlindquist/kitapp/blob/alpha/src/components/input.tsx

> Note: We _always_ work in the alpha channel. Then alpha pr -> beta pr -> main

**Reply by davelosert** [reply]
Thanks for the quick reply 🙂. Sounds great!

I'll have a look at the input, thanks for pointing me too it. 👍🏻

**Comment by johnlindquist** [comment]
@davelosert Added #562 to the latest build. Enjoy!

**Reply by davelosert** [reply]
That looks great! 🙂 
Thanks so much for putting it in that fast. I will definitly have a go at it later today.
